aviatrix/resource_aviatrix_controller_config.go

```diff
@@ -212,30 +230,39 @@ func resourceAviatrixControllerConfigCreate(d *schema.ResourceData, meta interfa
 	backupCloudType := d.Get("backup_cloud_type").(int)
 	backupAccountName := d.Get("backup_account_name").(string)
 	backupBucketName := d.Get("backup_bucket_name").(string)
+	backupStorageName := d.Get("backup_storage_name").(string)
+	backupContainerName := d.Get("backup_container_name").(string)
+	backupRegion := d.Get("backup_region").(string)
 	multipleBackups := d.Get("multiple_backups").(bool)
+
 	if backupConfiguration {
-		if backupCloudType == 0 || backupAccountName == "" || backupBucketName == "" {
-			return fmt.Errorf("please specify 'backup_cloud_type', 'backup_account_name' and 'backup_bucket_name'" +
-				" to enable backup configuration")
+		backupConfigErr := goaviatrix.ValidateBackupConfig(d)
```

Comment:
We should follow the pattern that is used throughout the code base and use the variable name `err` for error returns.